tvm/src/pass/stream_inference.cc

```diff
@@ -1273,20 +1277,25 @@ class StmtGrpReplacer final : public IRMutator {
 
 // 1. add annotation to kernel def node 
 // 2. mutate the producer marked with .new 
+// 3. remove defined but unsued vars
 class KernelAnnotator final : public IRMutator {
  public:
   KernelAnnotator(
     std::unordered_map<std::string, std::unordered_set<int>> map,
     std::unordered_map<std::string, Array<Expr>> mem_ports, 
-    Array<NodeRef>& api_args) :
-    arg_scope_map_(map), mem_ports_(mem_ports) {} 
+    std::unordered_set<const Variable*>& unused_vars) :
+    arg_scope_map_(map), mem_ports_(mem_ports), unused_vars_(unused_vars) {} 
 
   Stmt Mutate_(const Allocate* op, const Stmt& s) {
     Stmt stmt = IRMutator::Mutate_(op, s);
     op = stmt.as<Allocate>();
     std::string target_name = op->buffer_var.get()->name_hint;
-    if (target_name == "test") {
-      return this->Mutate(op->body);
+    // if (unused_vars_.count(op->buffer_var.get())) {
```

Comment:
Remove?

Reply:
Removed. I was trying to remove the allocated but unused buffers (e.g. allocate _top). But I found that it may incur some problems.